<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="goal_x" default="1.38" />
    <arg name="goal_y" default="2.08" />
    <!-- <arg name="goal_theta" default="2.08" /> -->
    <arg name="wall_offset" default="0.25" />

    <arg name="no_static_map" default="true" />

    <arg name="base_global_planner" default="navfn/NavfnROS" />
    <!-- <arg name="base_local_planner" default="dwa_local_planner/DWAPlannerROS" /> -->
    <arg name="base_local_planner" default="base_local_planner/TrajectoryPlannerROS"/>

    <node name="task" pkg="parc_phase2_solution" type="task_solution.py" args="$(arg goal_x) $(arg goal_y)" output="screen" />
    <node name="traffic_detector" pkg="parc_phase2_solution" type="traffic_detector.py" args="$(arg goal_x) $(arg goal_y)" output="screen" />
    <node name="target_calculator" pkg="parc_phase2_solution" type="target_calculator_node.py" args="$(arg wall_offset)" output="screen" />
    <node name="obstacle_detector" pkg="parc_phase2_solution" type="obstacles_detector.py" output="screen" />

    <node name="line_extractor" pkg="laser_line_extraction" type="line_extraction_node">
        <param name="~frequency" value="30" />
        <param name="~frame_id" value="base_scan" />
        <param name="~scan_topic" value="scan" />
        <param name="~publish_markers" value="true" />
        <param name="~bearing_std_dev" value="1e-5" />
        <param name="~range_std_dev" value="0.012" />
        <param name="~least_sq_angle_thresh" value="0.0001" />
        <param name="~least_sq_radius_thresh" value="0.0001" />
        <param name="~max_line_gap" value="0.5" />
        <param name="~min_line_length" value="0.7" />
        <param name="~min_range" value="0.1" />
        <param name="~max_range" value="250.0" />
        <param name="~min_split_dist" value="0.04" />
        <param name="~outlier_dist" value="0.06" />
        <param name="~min_line_points" value="10" />
    </node>

    <node pkg="move_base" type="move_base" respawn="false" name="move_base">

        <param name="base_global_planner" value="$(arg base_global_planner)" />
        <param name="base_local_planner" value="$(arg base_local_planner)" />
        <rosparam file="$(find parc_phase2_solution)/config/planner.yaml" command="load" />

        <!-- observation sources located in costmap_common.yaml -->
        <rosparam file="$(find parc_phase2_solution)/config/costmap_common.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find parc_phase2_solution)/config/costmap_common.yaml" command="load" ns="local_costmap" />

        <!-- local costmap, needs size -->
        <rosparam file="$(find parc_phase2_solution)/config/costmap_local.yaml" command="load" ns="local_costmap" />
        <param name="local_costmap/width" value="3.0" />
        <param name="local_costmap/height" value="3.0" />

        <!-- static global costmap, static map provides size -->
        <rosparam file="$(find parc_phase2_solution)/config/costmap_global_static.yaml" command="load" ns="global_costmap" unless="$(arg no_static_map)" />

        <!-- global costmap with laser, for odom_navigation_demo -->
        <rosparam file="$(find parc_phase2_solution)/config/costmap_global_laser.yaml" command="load" ns="global_costmap" if="$(arg no_static_map)" />
        <param name="global_costmap/width" value="100.0" if="$(arg no_static_map)" />
        <param name="global_costmap/height" value="100.0" if="$(arg no_static_map)" />
    </node>

</launch>
